app/scripts/lib/fxa-client.js

```diff
@@ -399,25 +403,31 @@ define(function (require, exports, module) {
         clientOptions.resume = options.resume;
       }
 
+      setMetricsContext(clientOptions, options);
+
       return client.passwordForgotResendCode(
         email,
         passwordForgotToken,
         clientOptions
       );
     }),
 
-    completePasswordReset: withClient((client, originalEmail, newPassword, token, code, relier) => {
+    completePasswordReset: withClient((client, originalEmail, newPassword, token, code, relier, options = {}) => {
       const email = trim(originalEmail);
 
-      return client.passwordForgotVerifyCode(code, token)
+      var clientOptions = {
+        keys: wantsKeys(relier),
+        sessionToken: true
+      };
+
+      setMetricsContext(clientOptions, options);
+
+      return client.passwordForgotVerifyCode(code, token, clientOptions)
```

Comment:
Is it alright that `clientOptions` is passed to `passwordForgotVerifyCode`? Will the auth server blow up? Maybe create a separate `clientOptions` for each of the calls to minimize assumptions about what each endpoint will accept.